src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SslOverTdsStream.cs

```diff
@@ -47,7 +49,50 @@ public void FinishHandshake()
         /// <param name="offset">Offset</param>
         /// <param name="count">Byte count</param>
         /// <returns>Bytes read</returns>
-        public override int Read(byte[] buffer, int offset, int count)
+        public override int Read(byte[] buffer, int offset, int count) =>
+            ReadInternal(buffer, offset, count, CancellationToken.None, false).GetAwaiter().GetResult();
```

Comment:
It looks like previously this would have called the synchronous Read methods on the underlying stream, but now it's invoking the asynchronous ones and then blocking on the result.  Why?  That's generally not a good direction to go.

Reply:
Ah, nevermind, I see you're switching in the implementation on using either the sync or async methods.  Ok.

Could you name the argument to ReadInternal so that it's clear what the bool means, e.g. `async: false` instead of just `false`?

Reply:
OK.